### ISSUE TITLE
Internal Error: DWARF - while using flexible array member

### ISSUE THREAD

**Issue by fafok29** [WhiteBox-Systems/whitebox#14]
### What happened?

I have a struct with a [flexible array member](https://gcc.gnu.org/onlinedocs/gcc/Zero-Length.html)
```
typedef struct {
	uint64_t len;
	uint8_t buf[]; <<< 
} arr_uint8_t;
```

Whenever I try to analyze a function that has arr_uint8_t type, I get the following error:

Actual behavior:
> Internal Error: DWARF
> Assertion failed in `dwarf_val_constant` - ../dwarf.h(1206):
>     form_val.kind == DW_VAL_constant
>     errno => 0: Success
> SIGTRAP hit in PID: 475436. Attach a debugger, hit 'x' to exit or 'c' to continue...

Expected behavior:

It should work with a flexible array member.

The same code but with zero length array doesn't produce any error
```
typedef struct {
	uint64_t len;
	uint8_t buf[0]; <<< 
} arr_uint8_t;
```

### System/app details

WhiteBox version - 0.91.1
WhiteBox commit  - a1735eaf840395556371fae40ad42c7c39b2d4a8
Assert level     - 5
Debug info       - false
Kernel           - Linux 5.15.0-46-generic
OS               - Ubuntu 22.04.1 LTS
Architecture     - x86-64
CPU name         - Intel(R) Core(TM) i7-8550U CPU @ 1.80GHz
CPU count        - 8
CPU MHz          - <unknown>
RAM              - 16G

### How to reproduce the error

Steps to reproduce:

0. rename algo1.txt to algo1.c
1. Start Whitebox
2. Open algo1.c in vim
3. Place cursor over fill_in_array function
4. call WhiteBoxConnect()
5. Get the error log mentioned in the description

If you change arr_uint8_t to use a zero-length array instead of a flexible array member and it will not assert.

Does the error happen consistently with the above steps?
Always

Minimal reproducible code example:
[algo1.txt](https://github.com/WhiteBox-Systems/whitebox/files/9303854/algo1.txt)

### Crash/log attachments & additional info

- gdb generates 13G of the core file, I will provide it if you won't be able to reproduce this issue.
- [whitebox_log.txt](https://github.com/WhiteBox-Systems/whitebox/files/9303878/whitebox_log.txt) 

Any additional information/comments:

I  noticed that the value of errno from a log is random
>     errno => 0: Success
>     errno => 2: No such file or directory
and so on

**Comment by azmr** [comment]
> I noticed that the value of errno from a log is random

Yeah it doesn't necessarily correspond to the source of the assertion failing, but it's occasionally useful as a debug prompt.

**Comment by azmr** [comment]
Fixed in `36b00c3`
